src/devdb/helpers.py

```diff
@@ -155,9 +156,19 @@
         "field_type": "boolean",
         "section": "n/a",
     },
+    "manual_hny_match_check": {
+        "description": "check if the manual HNY corrections are ingested and any HNY projects failed to be added will show up here",
+        "field_type": "boolean",
+        "section": "n/a",
+    },
+    "manual_corrections_not_applied": {
+        "description": "This will return the full list of jobs that have any manual corrections did not get applied.",
+        "field_type": "boolean",
+        "section": "n/a"
+    }
 }
 
-
+@st.cache
```

Comment:
do you know if this cache has a ttl? otherwise will show stale data if underlying data updates. I think its smart to think about where we can add caching, but it might be slightly more complex than this to create the right system for expiring it

Reply:
that's a good question @abrieff if I am understanding your questions/concerns correctly.

My understanding based on the [streamlit doc](https://docs.streamlit.io/library/api-reference/performance/st.cache) is that this is smart enough that the if any of the input change then the hash is recomputed and new value stored. But I guess since the branch is only input taken it might persists forever. There is ttl options maybe we can set it to something like 10 minutes (600 seconds)?

Reply:
Yes that's right, seems like a reasonable choice for now. It would be nice to increase the caching across this app to speed it up, have been noodling what that mechanism should be but haven't thought of anything better than time (we could probably retrieve the last updated at from boto3, which would be the best cache expiry but probably overkill for what this is)

Reply:
I see. I guess the config file some of our data product has from data library process is probably exactly what is needed for this kind of check? But without the implementation of that process on devdb maybe we could do with ttl. Let me know if the latest commit looks good and I will merge this in. Thanks!